### PR TITLE
Standardize on dt in update methods

### PR DESCRIPTION
**Pull request by spydon**
## Description

We used to have a mix of `t` and `dt` in the update methods, this PR standardizes the use of `dt`.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The Flame analyzer (`./scripts/analyze.sh`) does not report any problems on my PR.
- [x] I have added examples for new features in `doc/examples`.
- [x] I have formatted my code with `./scripts/format.sh`.
- [x] I read and followed the [Flutter Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I am willing to follow-up on review comments in a timely manner.
- [x] I am happy with the current version of this PR and it is ready to be reviewed
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

## Related Issues

#677

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning

### CODE REVIEW

**Review by erickzanardo — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
  - Add utility methods to the Anchor class to make it more "enum like"
  - Enable user-defined anchors
  - Added `toImage` method for the `Sprite` class
+ - Uniform use of dt instead of t in all update methods
```
Comment (wolfenrain) [inline CHANGELOG.md:25]:
```suggestion
 - Uniform use of `dt` instead of `t` in all update methods
```

File: lib/src/game/game.dart [hunk 2]
```diff
@@ -57,10 +57,10 @@ abstract class Game {
   /// It cannot be changed at runtime, because the game widget does not get rebuild when this value changes.
   Color backgroundColor() => const Color(0xFF000000);
 
-  /// Implement this method to update the game state, given that a time [t] has passed.
+  /// Implement this method to update the game state, given the time [dt] that has passed since the last update.
   ///
-  /// Keep the updates as short as possible. [t] is in seconds, with microseconds precision.
-  void update(double t);
+  /// Keep the updates as short as possible. [dt] is in seconds, with microseconds precision.
```
Comment (wolfenrain) [inline lib/src/game/game.dart:62]: Wait, isn't `dt` in milliseconds with microseconds precision? Or am I just too tired?

**Review by wolfenrain — APPROVED** [review]
LGTM, just one nit and a question :D

File: lib/src/game/game.dart (hunk 2 above)
Comment (spydon) [reply lib/src/game/game.dart:62]: You are too tired.

**Pull request merged**